File: examples/java/wrist-tempo/src/test/java/com/example/wrist_tempo/GreetingTest.java
package com.example.wrist_tempo;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

/**
 * Verifies the greeting assembly logic in {@link Greeting}.
 *
 * <p>Plain JVM JUnit 4 run by Maven Surefire ({@code mvn test}) -
 * no Android SDK, emulator, or Robolectric needed.</p>
 */
public class GreetingTest {

    @Test
    public void greetingIsAssembledFromProjectName() {
        // Greeting.greet() concatenates its argument, so this verifies
        // real logic rather than comparing two identical literals.
        assertEquals(
                "Hello from wrist-tempo!",
                Greeting.greet("wrist-tempo"));
    }

    @Test
    public void greetingReflectsAnArbitraryName() {
        assertEquals("Hello from wear!", Greeting.greet("wear"));
    }
}

File: examples/kotlin/wrist-counter/app/src/test/kotlin/com/example/wrist_counter/GreetingTest.kt
package com.example.wrist_counter

import org.junit.Assert.assertEquals
import org.junit.Test

/** Verifies the greeting interpolation logic in MainActivity.kt. */
class GreetingTest {
    @Test
    fun greetingIsAssembledFromProjectName() {
        // greeting() interpolates its argument, so this verifies real
        // logic rather than comparing two identical literals.
        assertEquals("Hello from wrist-counter!", greeting("wrist-counter"))
    }

    @Test
    fun greetingReflectsAnArbitraryName() {
        assertEquals("Hello from wear!", greeting("wear"))
    }
}
